aptos-move/framework/move-stdlib/src/natives/vector.rs

```diff
@@ -1,321 +0,0 @@
-#[derive(Debug, Clone)]
-pub struct SwapGasParameters {
-    pub base: InternalGas,
-}
-
-pub fn native_swap(
-    gas_params: &SwapGasParameters,
-    _context: &mut NativeContext,
-    ty_args: Vec<Type>,
-    mut args: VecDeque<Value>,
-) -> PartialVMResult<NativeResult> {
-    debug_assert!(ty_args.len() == 1);
-    debug_assert!(args.len() == 3);
-
-    let idx2 = pop_arg!(args, u64) as usize;
-    let idx1 = pop_arg!(args, u64) as usize;
-    let r = pop_arg!(args, VectorRef);
-    NativeResult::map_partial_vm_result_empty(
-        gas_params.base,
-        r.swap(idx1, idx2, &ty_args[0])
-            .map_err(native_error_to_abort),
-    )
-}
-
-pub fn make_native_swap(gas_params: SwapGasParameters) -> NativeFunction {
-    Arc::new(
-        move |context, ty_args, args| -> PartialVMResult<NativeResult> {
-            native_swap(&gas_params, context, ty_args, args)
-        },
-    )
-}
-
-fn native_error_to_abort(err: PartialVMError) -> PartialVMError {
-    let (major_status, sub_status_opt, message_opt, exec_state_opt, indices, offsets) =
-        err.all_data();
-    let new_err = match major_status {
-        StatusCode::VECTOR_OPERATION_ERROR => PartialVMError::new(StatusCode::ABORTED),
-        _ => PartialVMError::new(major_status),
-    };
-    let new_err = match sub_status_opt {
-        None => new_err,
-        Some(code) => new_err.with_sub_status(code),
-    };
-    let new_err = match message_opt {
-        None => new_err,
-        Some(message) => new_err.with_message(message),
-    };
-    let new_err = match exec_state_opt {
-        None => new_err,
-        Some(stacktrace) => new_err.with_exec_state(stacktrace),
-    };
-    new_err.at_indices(indices).at_code_offsets(offsets)
-}
-
-/***************************************************************************************************
- * module
- **************************************************************************************************/
-#[derive(Debug, Clone)]
-pub struct GasParameters {
-    pub empty: EmptyGasParameters,
-    pub length: LengthGasParameters,
-    pub push_back: PushBackGasParameters,
-    pub borrow: BorrowGasParameters,
-    pub pop_back: PopBackGasParameters,
-    pub destroy_empty: DestroyEmptyGasParameters,
-    pub swap: SwapGasParameters,
-}
-
-pub fn make_all(gas_params: GasParameters) -> impl Iterator<Item = (String, NativeFunction)> {
-    let natives = [
-        ("empty", make_native_empty(gas_params.empty)),
-        ("length", make_native_length(gas_params.length)),
-        ("push_back", make_native_push_back(gas_params.push_back)),
-        ("borrow", make_native_borrow(gas_params.borrow.clone())),
-        ("borrow_mut", make_native_borrow(gas_params.borrow)),
-        ("pop_back", make_native_pop_back(gas_params.pop_back)),
-        (
-            "destroy_empty",
-            make_native_destroy_empty(gas_params.destroy_empty),
-        ),
-        ("swap", make_native_swap(gas_params.swap)),
```

Comment:
how come you can delete this file

Reply:
Great finding. I should've mentioned this in the review guide. Here's the full story
0. A long time ago, the vector operations only existed in the form of native functions.
1. Later, we decided to make them bytecode instructions. The compiler would translate all calls to these functions into bytecodes. However, the original native function declarations & implementations were kept. (This was still Facebook-era)
2. Before our mainnet launch, I was looking into the sanity of our native functions, and realized these legacy functions still existed in our VM, so I went ahead and got them removed. Since we were still depending on the move-stdlib crate in the public Move repo at that time, I did this through a dirty hack -- filtering out the vector natives by name. This means, even today, the vector natives are not part of the VM.
  - https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/aptos-vm/src/natives.rs#L40
3. Later, I think Wolfgang discovered that due to a compiler bug, we had some code (on testnet?) that depended on these vector natives, so he tried to add them back, but for some reason I don't remember, the PR did not land. https://github.com/aptos-labs/aptos-core/pull/4788
4. A few months ago you forked the `move-stdlib` inside `aptos-core`
5. When I was working on this PR, I tried to convert the remaining natives to the safe variants, and discovered that I could not find the gas parameters for these vector natives. After some digging I realized they are not part of the VM, so I can safely delete them.

Reply:
If they are not used, I'm very happy to control-alt-delete them